From 3f29a92aef1b026d72772b81b791336e78f47309 Mon Sep 17 00:00:00 2001
From: Sukumar Ghorai <sukumar.ghorai@intel.com>
Date: Fri, 20 Feb 2015 15:19:53 +0530
Subject: [PATCH] adc-driver: early resume call added in S3-exit

Dollar Cove(X-power) GPADC Driver
When platform exit from S3 (mwait S3), acpi_thermal_resume path reads
temperature data from io-region. Which have dependency on adc-driver
and resume of adc-driver need to be done before thermal-resume.
Changed the resume call to resume_early.

Issue:https://jira01.devtools.intel.com/browse/GMINL-5644
Change-Id: I40debd2e44706b33f115c0ab93f1f588df41085b
Signed-off-by: Sukumar Ghorai <sukumar.ghorai@intel.com>
---
 drivers/iio/adc/dc_xpwr_gpadc.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/iio/adc/dc_xpwr_gpadc.c b/drivers/iio/adc/dc_xpwr_gpadc.c
index 5185008..634ac46 100644
--- a/drivers/iio/adc/dc_xpwr_gpadc.c
+++ b/drivers/iio/adc/dc_xpwr_gpadc.c
@@ -281,8 +281,8 @@ static int dc_xpwr_gpadc_resume(struct device *dev)
 #endif
 
 static const struct dev_pm_ops dc_xpwr_gpadc_pm_ops = {
-	.suspend	= dc_xpwr_gpadc_suspend,
-	.resume		= dc_xpwr_gpadc_resume,
+	.suspend_late = dc_xpwr_gpadc_suspend,
+	.resume_early = dc_xpwr_gpadc_resume,
 };
 
 static struct platform_driver dc_xpwr_gpadc_driver = {
-- 
1.9.1

